docs/dev/guidelines/server.rst

```diff
@@ -350,3 +350,240 @@ Please read `the AssertJ documentation <https://assertj.github.io/doc/#assertj-c
 
 
 Some parts of these guidelines are adapted from https://medium.com/@madhupathy/ultimate-clean-code-guide-for-java-spring-based-applications-4d4c9095cc2a
+
+22. General Testing Tips
```

Comment:
Most of this paragraph reads like it is detailing testing tips for writing tests for exercises (i.e. which give students feedback). I'm not sure if the `dev` documentation is the right place for that.